google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/cloud/AppEngineDeploymentConfigurator.java

```diff
@@ -115,6 +155,24 @@ public AppEngineDeploymentConfiguration createDefaultConfiguration(
     return new AppEngineDeploymentRunConfigurationEditor(
         project,
         source,
-        new CloudSdkAppEngineHelper(new File(server.getConfiguration().getCloudSdkHomePath())));
+        new CloudSdkAppEngineHelper(
+            project,
+            new File(server.getConfiguration().getCloudSdkHomePath()),
+            getEnvironment())
+    );
+  }
+
+  /**
+   * Lazily initialize the environment because this class is instantiated on IDE boot time
+   * and the project may not have yet been fully constructed.
+   */
+  public Environment getEnvironment() {
```

Comment:
Currently, it determined based on the project itself: if the project has the AE standard facet from the old plugin, then its standard. Otherwise, its flex. This came from the document that we went over earlier, but is obviously open for change.